syne_tune/optimizer/schedulers/searchers/searcher.py

```diff
@@ -516,13 +534,16 @@ def __init__(
         self,
         config_space,
         metric,
+        num_samples=None,
         points_to_evaluate=None,
         shuffle_config=True,
         **kwargs,
     ):
         super().__init__(config_space, metric, points_to_evaluate)
-        self._validate_config_space(config_space)
+        GridSearcher.DEFAULT_NSAMPLE = 5
```

Comment:
I think it would be cleaner to set `self.num_samples` to some default value if `num_samples` is None instead of introducing another constant

Reply:
self.num_samples is a dictionary, setting it to a default value requires going through hps in config_space, checking their type and if they are float or int, add self.num_samples[hp] = 5. I think it is cleaner to have a default constant independent of hps used in config_space. wdyt?

Reply:
I see, but in this case I would define the constant outside of the class